catchup/service.go

```diff
@@ -157,25 +157,31 @@ func (s *Service) IsSynchronizing() (synchronizing bool, initialSync bool) {
 	return
 }
 
+// triggerSync attempts to wake up the sync loop.
+func (s *Service) triggerSync() {
+	if syncing, initial := s.IsSynchronizing(); !syncing && !initial {
+		select {
+		case s.syncNow <- struct{}{}:
+		default:
+		}
```

Comment:
Without this there was a deadlock when catchup was wrapping up.

Reply:
It deadlocks because periodic sync isn't running when catchup is setting the sync round.